sstables/sstable_directory.cc

```diff
@@ -666,22 +661,28 @@ future<> sstable_directory::filesystem_components_lister::handle_sstables_pendin
 
     std::vector<future<>> futures;
 
-    co_await lister::scan_dir(pending_delete_dir, lister::dir_entry_types::of<directory_entry_type::regular>(), [this, &futures] (fs::path dir, directory_entry de) {
-        // push nested futures that remove files/directories into an array of futures,
-        // so that the supplied callback will not block scan_dir() from
-        // reading the next entry in the directory.
-        fs::path file_path = dir / de.name;
-        if (file_path.extension() == ".tmp") {
-            sstlog.info("Found temporary pending_delete log file: {}, deleting", file_path);
-            futures.push_back(remove_file(file_path.string()));
-        } else if (file_path.extension() == ".log") {
-            sstlog.info("Found pending_delete log file: {}, replaying", file_path);
-            auto f = replay_pending_delete_log(std::move(file_path));
-            futures.push_back(std::move(f));
-        } else {
-            sstlog.debug("Found unknown file in pending_delete directory: {}, ignoring", file_path);
+    sstlog.info("XEMUL LISTING {}", pending_delete_dir);
+    directory_lister lister(pending_delete_dir, lister::dir_entry_types::of<directory_entry_type::regular>());
+    co_await with_closeable(std::move(lister), [this, &dir = pending_delete_dir, &f = futures] (directory_lister& lister) -> future<> {
+        auto& pending_delete_dir = dir;
+        auto& futures = f;
```

Comment:
ditto, I think this can be collected inside the coroutine lambda and be co_return'ed to the caller.